### ISSUE TITLE
[pause] Refactor state

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#4556]
We can clean some things up here:

- stop storing pauseInfo
- store why directly

**Comment by jasonLaster** [comment]
here are some notes

```diff
diff --git a/src/actions/pause/paused.js b/src/actions/pause/paused.js
index d13b917..f2602b2 100644
--- a/src/actions/pause/paused.js
+++ b/src/actions/pause/paused.js
@@ -27,11 +27,9 @@ export function paused(pauseInfo: Pause) {
     const { frames, why, loadedObjects } = pauseInfo;
 
     const mappedFrames = await updateFrameLocations(frames, sourceMaps);
-    const frame = mappedFrames[0];
 
     dispatch({
       type: "PAUSED",
-      pauseInfo: { why, frame, frames },
       frames: mappedFrames,
       selectedFrameId: frame.id,
       loadedObjects: loadedObjects || []
@@ -46,6 +44,7 @@ export function paused(pauseInfo: Pause) {
       dispatch(evaluateExpressions());
     }
 
+    const frame = mappedFrames[0];
     const { line, column } = frame.location;
     await dispatch(selectSource(frame.location.sourceId, { line, column }));
 
diff --git a/src/reducers/pause.js b/src/reducers/pause.js
index 03f77a8..b8dbe86 100644
--- a/src/reducers/pause.js
+++ b/src/reducers/pause.js
@@ -17,7 +17,7 @@ import { isEmpty } from "lodash";
 import type { Action } from "../actions/types";
 
 type PauseState = {
-  pause: ?any,
+  why: ?any,
   isWaitingOnBreak: boolean,
   frames: ?(any[]),
   frameScopes: any,
@@ -30,9 +30,9 @@ type PauseState = {
 };
 
 export const State = (): PauseState => ({
-  pause: undefined,
   isWaitingOnBreak: false,
   frames: undefined,
+  why: undefined,
   selectedFrameId: undefined,
   frameScopes: {},
   loadedObjects: {},
@@ -45,8 +45,7 @@ export const State = (): PauseState => ({
 function update(state: PauseState = State(), action: Action): PauseState {
   switch (action.type) {
     case "PAUSED": {
-      const { selectedFrameId, frames, loadedObjects, pauseInfo } = action;
-      pauseInfo.isInterrupted = pauseInfo.why.type === "interrupted";
+      const { selectedFrameId, frames, loadedObjects, why } = action;
 
       // turn this into an object keyed by object id
       const objectMap = {};
@@ -56,8 +55,8 @@ function update(state: PauseState = State(), action: Action): PauseState {
 
       return Object.assign({}, state, {
         isWaitingOnBreak: false,
-        pause: pauseInfo,
         selectedFrameId,
+        why,
         frames,
         frameScopes: {},
         loadedObjects: objectMap
@@ -81,12 +80,7 @@ function update(state: PauseState = State(), action: Action): PauseState {
     case "TOGGLE_PRETTY_PRINT":
       if (action.status == "done") {
         const frames = action.value.frames;
-        const pause = state.pause;
-        if (pause) {
-          pause.frame = frames[0];
-        }
-
-        return Object.assign({}, state, { pause, frames });
+        return Object.assign({}, state, { frames });
       }
 
       break;
@@ -275,6 +269,11 @@ export function getDebuggeeUrl(state: OuterState) {
   return state.pause.debuggeeUrl;
 }
 
+// lets use this to decide if we're interupted why.type === "interrupted";
+export function getPauseReason(state: OuterState) {
+  return state.why;
+}
+
 // NOTE: currently only used for chrome
 export function getChromeScopes(state: OuterState) {
   const frame = getSelectedFrame(state);
diff --git a/src/test/mochitest/browser_dbg-minified.js b/src/test/mochitest/browser_dbg-minified.js
index 67a570f..b29d1f5 100644
--- a/src/test/mochitest/browser_dbg-minified.js
+++ b/src/test/mochitest/browser_dbg-minified.js
@@ -1,5 +1,5 @@
 /* Any copyright is dedicated to the Public Domain.
- * http://creativecommons.org/publicdomain/zero/1.0/ */
+* http://creativecommons.org/publicdomain/zero/1.0/ */
 
 // Tests minfied + source maps.
 
```

**Comment by jasonLaster** [comment]
Really happy to be able to do this refactor. Cleaning up the pause store has been a goal of mine for awhile and I think this is significantly simpler.

A little bit of historical context:

When we were initially building the Breakpoints pane, we wanted to show if the app was currently paused at a breakpoint. The outlier for showing pause locations are interrupts where we dont store the frame or location. To make it easy to tell if we were interrupted we added an extra field `isInterrupted` in the store. This was the only time i can think of where we added a computed field. Since then we have used selectors and utils more effectively and this has felt like a code smell.

Another decision we made early on was to store the pause packet (top frame, why, ...) in its own field. In hindsight, top frame should have been a selector and why should have been its own field.
